packages/rich-text/src/component/index.js

```diff
@@ -248,6 +248,12 @@ export function useRichText( {
 
 	return {
 		value: record.current,
+		// A function to get the most recent value so event handlers in
+		// useRichText implementations have access to it. For example when
+		// listening to input events, we internally update the state, but this
+		// state is not yet available to the input event handler because React
+		// may re-render asynchronously.
+		getValue: () => record.current,
```

Comment:
Would the same work if instead of passing `ref.current` or `() => ref.current`, we just pass `ref` here and access `ref.current` in the event handlers?

Reply:
Yes, but it might get confusing if that ref is passed to another ref? I also think the naming of `getValue` is clearer than `valueRef.current` or something like that.

Reply:
> but it might get confusing if that ref is passed to another ref?

What do you mean by this?

Personally, I would prefer `ref.current` as it is just the standard idiomatic React's ref. But I also wouldn't mind `getValue()` as it's also simple and clear enough 👍 .

Reply:
We're passing all props within a ref here, so it's a nested ref: https://github.com/WordPress/gutenberg/pull/48210/files#diff-8ba153e0dccbf25a179a794cb63aacd09e575d5dcfa819f8aefa33086ddbbb6eR52